assets/js/modules/adsense/dashboard/dashboard-widget-performance.js

```diff
@@ -76,17 +88,6 @@ class AdSensePerformanceWidget extends Component {
 					context: [ 'Single', 'Dashboard' ],
 					callback: ( result ) => {
 
-						// If there are no impressions, the site is not yet displaying ads.
-						if (
-							result && result.totals &&
-							(
-								null === result.totals[2] ||
-								0    === result.totals[2]
-							)
-						) {
-							handleZeroData();
-						}
-
```

Comment:
Shouldn't this check remain here as well (and thus be present in both locations, for both requests)? I'd expect both to `handleZeroData()` as necessary.

Reply:
@felixarntz  this should be only present in `earning-28days`, as for sites with newly AdSense setup would not have data for prev 28 days, adding this logic to the `earning-prev28days` would end up setting zeroData and therefor not rendering data at all.

Reply:
Additionally noting: the purpose of this check is to see if we are getting data from the API. We chose the "impressions" data because it is very likely that if you have AdSense installed correctly you will have impressions (while other values might be 0 even though you do have data). The bug here was that we were looking at the previous 28 days vs the (current) last 28 days data.